jiant/tasks/lib/winogrande.py

```diff
@@ -0,0 +1,123 @@
+            prompt, choice = truncate_sequences(
+                tokens_ls=[self.prompt, choice],
+                max_length=feat_spec.max_seq_length - special_tokens_count,
+            )
+            unpadded_inputs = add_cls_token(
+                unpadded_tokens=(
+                    # choice
+                    choice
+                    + [tokenizer.sep_token]
+                    + maybe_extra_sep
+                    # prompt
+                    + prompt
+                    + [tokenizer.sep_token]
+                ),
+                unpadded_segment_ids=(
+                    # choice
+                    [feat_spec.sequence_a_segment_id] * (len(choice) + 1)
+                    + maybe_extra_sep_segment_id
+                    # prompt + sep
+                    + [feat_spec.sequence_b_segment_id] * (len(prompt) + 1)
+                ),
+                tokenizer=tokenizer,
+                feat_spec=feat_spec,
+            )
+            input_set = create_input_set_from_tokens_and_segments(
+                unpadded_tokens=unpadded_inputs.unpadded_tokens,
+                unpadded_segment_ids=unpadded_inputs.unpadded_segment_ids,
+                tokenizer=tokenizer,
+                feat_spec=feat_spec,
+            )
+            input_set_ls.append(input_set)
+            unpadded_inputs_ls.append(unpadded_inputs)
+
+        return DataRow(
+            guid=self.guid,
+            input_ids=np.stack([input_set.input_ids for input_set in input_set_ls]),
+            input_mask=np.stack([input_set.input_mask for input_set in input_set_ls]),
+            segment_ids=np.stack([input_set.segment_ids for input_set in input_set_ls]),
+            label_id=self.label_id,
+            tokens_list=[unpadded_inputs.unpadded_tokens for unpadded_inputs in unpadded_inputs_ls],
+        )
+
+
+@dataclass
+class DataRow(mc_template.DataRow):
+    pass
+
+
+@dataclass
+class Batch(mc_template.Batch):
+    pass
+
+
+class WinograndeTask(mc_template.AbstractMultipleChoiceTask):
+    Example = Example
+    TokenizedExample = Example
+    DataRow = DataRow
+    Batch = Batch
+
+    CHOICE_KEYS = [1, 2]
+    CHOICE_TO_ID, ID_TO_CHOICE = labels_to_bimap(CHOICE_KEYS)
+    NUM_CHOICES = len(CHOICE_KEYS)
+
+    def get_train_examples(self):
+        return self._create_examples(lines=read_json_lines(self.train_path), set_type="train")
+
+    def get_val_examples(self):
+        return self._create_examples(lines=read_json_lines(self.val_path), set_type="val")
+
+    def get_test_examples(self):
+        return self._create_examples(lines=read_json_lines(self.test_path), set_type="test")
+
+    @classmethod
+    def _create_examples(cls, lines, set_type):
+        examples = []
+
+        for i, line in enumerate(lines):
+            choice_pre, option = line["sentence"].split("_")
+
+            examples.append(
+                Example(
+                    guid="%s-%s" % (set_type, i),
+                    prompt=option,
```

Comment:
{"qID": "3FCO4VKOZ4BJQ6IFC0VAIBK4KTWE7U-2", "sentence": "Sarah was a much better surgeon than Maria so _ always got the easier cases.", "option1": "Sarah", "option2": "Maria", "answer": "2"}

I can tell the outcome will be the same, but is the choice and prompt in TokenizedExample and here both reversed? And the variable name option is not very precise.

Reply:
Yes, they have been reversed in TokenizedExample as well. This is true about the variable name option... Let me know if I should change this in another push.